### PR TITLE
Pass ctx to Validate, ValidateUpdate

### PR DESCRIPTION
**Pull request by ncdc**
Pass ctx to Validate/ValidateUpdate. This is useful so strategies can
access data such as the current user.

### CODE REVIEW

**Comment by ncdc** [comment]
@smarterclayton @thockin PTAL

**Comment by vmarmol** [comment]
Assigning to @smarterclayton, feel free to re-assign.

**Comment by ncdc** [comment]
@smarterclayton actually, now that I look at the function signature, since PrepareForCreate/Update doesn't return anything right now, we won't be able to stop a create/update from proceeding when there's a referential integrity issue (i.e. not authorized for Bob to access Alice's ImageStream).

**Comment by ncdc** [comment]
Does it make sense to have these methods return a ValidationErrorList?

**Comment by smarterclayton** [comment]
Maybe an error.  However, that gets into the question of why wouldn't this then be folded into Validate, where there is just two parts of validation in the strategy - inherent, and referential.

**Comment by smarterclayton** [comment]
Having Prepare* return an error seems to just be duplicating Validate*

**Comment by ncdc** [comment]
I thought you had said you didn't want to do referential integrity checking in the Validate* methods. I'm quite happy to modify Validate and ValidateUpdate to take in the ctx instead.

**Comment by smarterclayton** [comment]
Not the core Validate methods - the Strategy validate methods.

**Comment by ncdc** [comment]
@smarterclayton updated to pass ctx to the Validate* strategy methods

File: pkg/registry/controller/rest.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ package controller
 import (
 	"fmt"
 
+	"strconv"
```
Comment (smarterclayton) [inline pkg/registry/controller/rest.go]: Nuke the whitespace above this

**Comment by smarterclayton** [comment]
After you fix that one line LGTM

File: pkg/registry/controller/rest.go (hunk 1 above)
Comment (ncdc) [inline pkg/registry/controller/rest.go]: That's from goimports, but ok

Comment (ncdc) [inline pkg/registry/controller/rest.go]: Fixed

**Comment by smarterclayton** [comment]
Goimports isn't smart enough to know that "fmt" and "strconv" are in the same group, it just knows that strconv doesn't belong with the deeply nested types.

**Pull request merged**